torch_geometric/loader/mixin.py

```diff
@@ -71,6 +74,86 @@ def __call__(self, worker_id):
             self.func(worker_id)
 
 
+class MemMixin:
+    r"""A context manager to enable logging of dataloder workers
+    memory consumption.
+    """
+    def mem_init_fn(self, worker_id):
+        logging.debug(
+            f"woker {worker_id} process is using {round((psutil.Process(                os.getpid()).memory_info().rss / 1024 ** 2),2)} MB of memory"
+        )
+        self.worker_init_fn_old(worker_id)
+
+    @contextmanager
+    def enable_memory_log(self):
+        self.worker_init_fn_old = WorkerInitWrapper(self.worker_init_fn)
+        try:
+            self.worker_init_fn = self.mem_init_fn
+            yield
+        finally:
+            self.worker_init_fn = self.worker_init_fn_old
+
+
+class MultithreadMixin:
+    r"""A context manager to enable multithreading in dataloder workers.
+    It changes the default value of threads used in the sampler from 1 to `worker_threads`.
+    """
+    def mt_init_fn(self, worker_id):
+        try:
+            torch.set_num_threads(int(self.worker_threads))
+        except IndexError:
+            raise ValueError(f"Cannot set multithreading for {worker_id}")
+        self.worker_init_fn_old(worker_id)
+
+    @contextmanager
+    def enable_multithreading(
+            self, worker_threads: Optional[int] = torch.get_num_threads() / 2):
+        r"""Enables multithreading in worker subprocess.
+
+        Args:
+            worker_threads ([int], optional): Number of cores for sampling.
+            By defualt it uses half of all avaialbe CPU cores
+            (default: : `torch.get_num_threads() / 2`)
```

Comment:
Should the default heuristic also be based on `num_workers`, to avoid oversubscription?

Reply:
You're right, I've changed it to: 
```
self.worker_threads = int(torch.get_num_threads() / self.num_workers)
```